src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/NullBeforeInstanceOfCheck.java

```diff
@@ -0,0 +1,144 @@
+ * </p>
+ * <ul>
+ * <li>
+ * Property {@code tokens} - tokens to check
+ * Default value is:
+ * <a href="https://checkstyle.org/apidocs/com/puppycrawl/tools/checkstyle/api/TokenTypes.html#LITERAL_IF">
+ * LITERAL_IF</a>.
+ * </li>
+ * </ul>
+ * <p>
+ * To configure the check:
+ * </p>
+ * <pre>
+ * &lt;module name=&quot;NullBeforeInstanceOf&quot;/&gt;
+ * </pre>
+ * <p>Example:</p>
+ * <pre>
+ * class Test {
+ *
+ *     if (myObj != null &amp;&amp; myObj instanceof Object) { // violation
+ *
+ *     }
+ *
+ *     if (myObj != null) { // violation
+ *         if (myObj instanceof Object) {
+ *
+ *         }
+ *     }
+ *
+ *    if ((myObj != null) &amp;&amp; (myObj instanceof Object)) {}  // violation
+ * }
+ * </pre>
+ * @since 8.33
+ */
+@StatelessCheck
+public class NullBeforeInstanceOfCheck extends AbstractCheck {
+
+    /**
+     * A key is pointing to the warning message null.before.instanceof in "messages.properties"
+     * file.
+     */
+    public static final String MSG_KEY = "null.before.instanceof";
+
+    @Override
+    public int[] getDefaultTokens() {
+        return getAcceptableTokens();
+    }
+
+    @Override
+    public int[] getAcceptableTokens() {
+        return new int[] {
+            TokenTypes.LITERAL_IF,
+        };
+    }
+
+    @Override
+    public int[] getRequiredTokens() {
+        return CommonUtil.EMPTY_INT_ARRAY;
+    }
+
+    @Override
+    public void visitToken(DetailAST ast) {
+        final DetailAST landNode = ast.findFirstToken(TokenTypes.EXPR)
+                .findFirstToken(TokenTypes.LAND);
+        if (landNode == null) {
+            if (checkOnNextLine(ast)) {
+                log(ast, MSG_KEY);
+            }
+        }
+        else {
+            final DetailAST notEqualNode = landNode.findFirstToken(TokenTypes.NOT_EQUAL);
+            final DetailAST instanceOfNode =
+                    landNode.findFirstToken(TokenTypes.LITERAL_INSTANCEOF);
+            if (notEqualNode != null && instanceOfNode != null
+                    && getText(notEqualNode).equals(getText(instanceOfNode))) {
+                log(ast, MSG_KEY);
+            }
+        }
+    }
+
+    private static String getText(DetailAST ast) {
+        return ast.findFirstToken(TokenTypes.IDENT).getText();
+    }
+
+    /**
+     * Check if there is a unnecessary 'null' check before 'instanceof' expression when
+     * 'instanceof' expression is on next line of 'null' check.
+     *
+     * @param ast DetailAST ast
+     * @return error boolean
+     */
+    private static boolean checkOnNextLine(DetailAST ast) {
+        final DetailAST notEqualNode = ast.findFirstToken(TokenTypes.EXPR)
+                .findFirstToken(TokenTypes.NOT_EQUAL);
+        boolean error = false;
+        if (notEqualNode != null
+                && notEqualNode.findFirstToken(TokenTypes.LITERAL_NULL) != null) {
+            final DetailAST slistNode = ast.findFirstToken(TokenTypes.SLIST);
```

Comment:
please check if the following test will pass or fail:
```
        if (myObj != null) // violation
            if (myObj instanceof Object)
                System.out.println();
```

Reply:
@timurt  I test it, and it is caught by NullBeforeInstanceOfCheck
cat Test3.java
```java
package checkstyle_demo.NullBeforeInstanceOfCheck;

public class Test3 {
    public void main(int[] obj) {
        int[] myObj = obj;
        if (myObj != null) { // violation
            if (myObj instanceof Object)
                System.out.println();
        }
    }
}
```
cat config.xml
```xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
        "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
        "https://checkstyle.org/dtds/configuration_1_3.dtd">
<module name="Checker">
    <module name="TreeWalker">
        <module name="NullBeforeInstanceOfCheck"/>
    </module>
</module>
```
Starting audit...
[ERROR] /var/tmp/checkstyle_demo/NullBeforeInstanceOfCheck/Test3.java:6:9: Null check is redundant when using instanceOf. [NullBeforeInstanceOf]
Checkstyle ends with 1 errors.
Audit done.

Reply:
all these examples are considered and are in test cases.

Reply:
No, without curly braces:

```
if (myObj != null) 
     if (myObj instanceof Object) System.out.println();
```

Reply:
sorry,  this is blind spots, I forgot to consider this example

Reply:
done, I consider this case now